src/store/Account.ts

```diff
@@ -584,24 +578,40 @@ export default {
                     .getMultisigAccountGraphInfo(currentAccountAddress)
                     .toPromise();
 
-                const currentMultisigAccountsInfo = MultisigService.getMultisigInfoFromMultisigGraphInfo(currentMultisigAccountGraphInfo);
-                const rootAddressIndex = currentMultisigAccountsInfo.length - 1;
-                const rootAddress = currentMultisigAccountsInfo[rootAddressIndex].accountAddress;
+                const rootMultisigAccountGraph = new Map(currentMultisigAccountGraphInfo.multisigEntries);
+
+                for (const [currentLevel, multisigAccountInfos] of currentMultisigAccountGraphInfo.multisigEntries) {
+                    if (currentLevel < 0) {
+                        for (const multisigAccountInfo of multisigAccountInfos) {
+                            const fetchedMultisigAccountGraphInfo = await repositoryFactory
+                                .createMultisigRepository()
```

Comment:
use forEach ?

Reply:
Do you mean map it into array of `Promise`s?

Reply:
nevermind, looking again, i think what you have is fine 👍